edk2toollib/uefi/edk2/parsers/base_parser.py

```diff
@@ -266,27 +285,45 @@ def ProcessConditional(self, text):
         tokens = text.split()
         if(tokens[0].lower() == "!if"):
             # need to add support for OR/AND
-            if(len(tokens) < 4):
+            if (len(tokens) == 2):
+                value = self.ConvertToInt(tokens[1].strip())
+                self.PushConditional(value == 1)  # if the value is true
+            # we can have tokens in 4, 8, 12 etc
+            elif len(tokens) >= 4 and len(tokens) % 4 == 0:
+                con = self.ComputeResult(tokens[1].strip(), tokens[2].strip(), tokens[3].strip())
```

Comment:
Right if it's more than 4 and in multiple of 4 then we can parse it. We need to implement a proper parser with maybe a stack since we need to support () and other things

Reply:
if TEST1 == TEST2 AND test3 == test4 is 8 tokens
if TEST1 == TEST2 AND test3 == test4 OR test5 == test6 is 8 tokens
and so on

Reply:
I didn't add support for +4 tokens but made sure it didn't crash on longer statements